src/backend/base/langflow/components/prototypes/UpdateData.py

```diff
@@ -1,40 +1,119 @@
-from langflow.custom import CustomComponent
+from typing import Any
+
+from langflow.custom import Component
+from langflow.inputs.inputs import IntInput, MessageTextInput, DictInput, DataInput, BoolInput
+from langflow.io import Output
+
+from langflow.field_typing.range_spec import RangeSpec
 from langflow.schema import Data
+from langflow.schema.dotdict import dotdict
 
 
-class UpdateDataComponent(CustomComponent):
-    display_name = "Update Data"
-    description = "Update Data with text-based key/value pairs, similar to updating a Python dictionary."
-    name = "UpdateData"
-
-    def build_config(self):
-        return {
-            "data": {
-                "display_name": "Data",
-                "info": "The record to update.",
-            },
-            "new_data": {
-                "display_name": "New Data",
-                "info": "The new data to update the record with.",
-                "input_types": ["Text"],
-            },
-        }
-
-    def build(
-        self,
-        data: Data,
-        new_data: dict,
-    ) -> Data:
-        """
-        Updates a record with new data.
+class UpdateOrAppendDataComponent(Component):
+    display_name: str = "Update or Append Data"
+    description: str = "Dynamically update or append data with a specified number of fields."
+    name: str = "UpdateData"
 
-        Args:
-            record (Data): The record to update.
-            new_data (dict): The new data to update the record with.
+    inputs = [
+        DataInput(
+            name="old_data",
+            display_name="Data",
+            info="The record to update.",
+        ),
+        IntInput(
```

Comment:
It would be nice if we didn't need to do this, and the user could just click a `+` or hit enter to add a new field.

Reply:
Did you mean the number of fields ?

The create data had the functionality  to  +/- the number of fields, I took the same approach in update data.
Yes the user can +/- the required number of fields.